cpp/include/cudf/join.hpp

```diff
@@ -327,10 +330,25 @@ class hash_join {
    *
    * @param has_nulls Flag to indicate if there exists any nulls in the `build` table or
    *        any `probe` table that will be used later for join
+   *
+   * @deprecated Deprecated constructor. Use the new load factor overload instead.
+   */
+  hash_join(cudf::table_view const& build,
+            nullable_join has_nulls,
+            null_equality compare_nulls,
+            rmm::cuda_stream_view stream = cudf::get_default_stream());
+
+  /**
+   * @copydoc hash_join(cudf::table_view const&, null_equality, rmm::cuda_stream_view)
+   *
+   * @param has_nulls Flag to indicate if there exists any nulls in the `build` table or
+   *        any `probe` table that will be used later for join
```

Comment:
```suggestion
```

Reply:
I meant this `has_nulls` param is not added here, so to `copydoc` should already cover it. Here we only need to add docs for the new `load_factor` param.

Reply:
I see what you mean. The `has_nulls` parameter was first introduced in a deprecated API, but since we can't copy documentation from deprecated APIs, We have to use the documentation from `hash_join(cudf::table_view const&, null_equality, rmm::cuda_stream_view)` as a base and manually add documentation for both `has_nulls` and `load_factor`.